### PR TITLE
server: clear v2 form in suspended state

### PR DESCRIPTION
**Pull request by d0s056f**

### CODE REVIEW

File: server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java [hunk 1]
```diff
@@ -142,6 +142,8 @@ public FormSubmitResult submit(ProcessKey processKey, String formName, Map<Strin
             Map<String, Object> args = new LinkedHashMap<>();
             args.put(form.name(), new LinkedHashMap<>(data));
 
+            formManager.delete(processKey, form.name());
```
Comment (ibodrov) [inline server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java:145]: If I understand it correctly, the form removal is already happening here https://github.com/walmartlabs/concord/blob/6e6a2e5189f79482dfff1fbb74e108f599ddf7d0/server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java#L182

Comment (d0s056f) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java:145]: what you shared is happening after the process is resumed. form logic is breaking with below error if removal does not happen on form submit -
```
14:29:40 [INFO ] Process status: RUNNING
java.lang.IllegalStateException: Can't find eventRef: 8853047f-84f4-48d2-995e-15ab18a7f3ec
	at com.walmartlabs.concord.svm.VM.resume(VM.java:68)
	at com.walmartlabs.concord.runtime.v2.runner.Runner.resume(Runner.java:99)
	at com.walmartlabs.concord.runtime.v2.runner.Main.resume(Main.java:237)
	at com.walmartlabs.concord.runtime.v2.runner.Main.execute(Main.java:151)
	at com.walmartlabs.concord.runtime.v2.runner.Main.main(Main.java:94)
14:29:41 [ERROR] Process exit code: 1
14:29:41 [INFO ] Process status: FAILED
```

Comment (ibodrov) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java:145]: AFAIK, you and @brig are working on a fix for the same thing. @brig, is that so?

Comment (brig) [reply server/impl/src/main/java/com/walmartlabs/concord/server/process/form/FormServiceV2.java:145]: @ibodrov yep :)

**Comment by ibodrov** [comment]
@concord-ci test it

**Review by ibodrov — APPROVED** [review]

**Pull request merged**